What is a List? 

A list is collection of elements that are stored in a variable. List can be like arrays. It is Ordered and Mutable and can contain items of different data types. Also, it can store Duplicate Items. It is Contained in [] brackets. List can add, Remove and delete items.

What is a Tuple?

A tuple is a collection of items which are like a list, but it is immutable.  It is also ordered collection and can store duplicate values as well as different datatype items. It is contained in () brackets. Tuples cannot add, remove or updated items.

What is the difference between List and Tuple?

List is a collection of items which can be mutable, but Tuples are immutable. Lists have syntax as [] whereas tuples have () brackets. Tuple is Faster than lists.

Python Program to find the largest element in the list.

In [ ]:
a = [2,3,4,5,8,1,0]
largest = a[0]
for i in range(1, len(a)):
    if(a[i] > largest):
        largest = a[i]
print("Largest element is:", largest)


Largest element is: 8


Python program to interchange first and last elements in a list

In [4]:
a = [1,2,3,4,5,6,8]

t = a[0]
a[0] = a[-1]
a[-1] = t

print(a)

[8, 2, 3, 4, 5, 6, 1]


Python program to swap two elements in a list

In [5]:
a = [1,3,45,6,23,2]

s1 = int(input("Enter the first position of the element to be swapped: "))
s2 = int(input("Enter the second position of the element to be swapped: "))
t = a[s1]
a[s1] = a[s2]
a[s2] = t
print(a)

[1, 3, 23, 6, 45, 2]


Python program to Reverse a List

In [1]:
a = [1,2,3,4,5,6,7,8,9]

for i in range(len(a)//2):
   a[i], a[len(a)-1-i] = a[len(a)-1-i], a[i]
print(a)

[9, 8, 7, 6, 5, 4, 3, 2, 1]


Python program to count occurrences of an element in a list

In [2]:
a = [1,2,3,4,5,6,7,8,9]
print("Original list:", a)
searchItem = int(input("Enter the element to be searched: "))

for i in range(len(a)):
    if (a[i] == searchItem):
        print("Element found at index:", i)
        break


Original list: [1, 2, 3, 4, 5, 6, 7, 8, 9]
Element found at index: 4


Python program to find the sum of elements in a list

In [4]:
a = [1,2,3,4,5,6,7,8,9]
sum  = 0
for i in range(len(a)):
    sum += a[i]
print("Sum of all elements in the list is:", sum)

Sum of all elements in the list is: 45


Python program to Multiply all numbers in the list

In [7]:
a = [1,2,3]
multiply = 1
for i in range(len(a)):
    multiply *= a[i]
print("Multiplication of all elements in the list is:", multiply)

Multiplication of all elements in the list is: 6


What are the ways to find the length of a list ?

len() is one function that we can get the length of an array. Otherwise, we can use for loops to create a function to identify the length from the index.

Python program to find the smallest and largest number in a list (Without min-max function)

In [8]:
a = [1,2,3,4,5,6,7,8,9]
smallest = a[0]
largest = a[0]
for i in range(len(a)):
    if a[i] < smallest:
        smallest = a[i]
    if a[i] > largest:
        largest = a[i]
print("Smallest element is:", smallest)
print("Largest element is:", largest)

Smallest element is: 1
Largest element is: 9


Python Program to find the area of a circle

In [9]:
import math
rad = int(input("Enter the radius of the circle: "))
area = math.pi * rad * rad
print("Area of the circle is:", area)

Area of the circle is: 28.274333882308138


Take inputs from the user to make a list. Again take one input from the user and delete that element from a list.

In [17]:
n  = int(input("Enter the Number of elements in the list: "))
a = []
for i in range(n):
    b = int(input("Enter the element: "))
    a.append(b)

dele = int(input("Enter the element to be deleted: "))

for i in range(len(a)):
    if(a[i] == dele):
        del a[i]
        break
print ("List after deletion of the element is:", a)

List after deletion of the element is: [1, 2, 4, 5]


You are given a list of integer elements. Make a new list that will store a square of elements of the previous list. (With and without list comprehension) ● Input_list = [2,5,6,12] ● Output_list = [4,25,36,144]

In [19]:
n  = int(input("Enter the Number of elements in the list: "))
a = []
for i in range(n):
    b = int(input("Enter the element: "))
    a.append(b)
b = []
for i in range(n):
    t = a[i] ** 2
    b.append(t)

print ("List after squaring the elements is:", b)

List after squaring the elements is: [4, 9, 16]


WAP to create two lists, one containing all even numbers and the other containing all odd numbers between 0 to 151

In [20]:
evenlist = []
oddlist = []
for i in range(1, 152):
    if(i%2 == 0):
        evenlist.append(i)
    if(i%2 != 0):
        oddlist.append(i)
print("List of even numbers from 1 to 151 is:", evenlist)
print("List of odd numbers from 1 to 151 is:", oddlist)

List of even numbers from 1 to 151 is: [2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100, 102, 104, 106, 108, 110, 112, 114, 116, 118, 120, 122, 124, 126, 128, 130, 132, 134, 136, 138, 140, 142, 144, 146, 148, 150]
List of odd numbers from 1 to 151 is: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151]


Python program to Count Even and Odd numbers in a List

In [26]:
even = []
odd = []
for i in range(150):
    if(i%2 == 0):
        even.append(i)
    else:
        odd .append(i)
print("Sum of even numbers from 1 to 151 is:", len(even))
print("Sum of odd numbers from 1 to 151 is:", len(odd))

Sum of even numbers from 1 to 151 is: 75
Sum of odd numbers from 1 to 151 is: 75


WAP to make new lists, containing only numbers which are divisible by 4, 6, 8, 10, 3, 5, 7, and 9 in separate lists for range(0,151)

In [29]:
a = []
for i in range(1,152):
    a.append(i)
div4 =[]
div6 =[]
div8 =[]
div10 =[]
div3 =[]
div5 =[]
div7 =[]

print("List of numbers from 1 to 151 is:", a)
for i in range(len(a)):
    if(a[i]%3 == 0):
        div3.append(a[i])
    if(a[i]%4 == 0):
        div4.append(a[i])
    if(a[i]%5 == 0):
        div5.append(a[i])
    if(a[i]%6 == 0):
        div6.append(a[i])
    if(a[i]%7 == 0):
        div7.append(a[i])
    if(a[i]%8 == 0):
        div8.append(a[i])
    if(a[i]%10 == 0):
        div10.append(a[i])
print("List of numbers divisible by 3 is:", div3)
print("List of numbers divisible by 4 is:", div4)
print("List of numbers divisible by 5 is:", div5)
print("List of numbers divisible by 6 is:", div6)
print("List of numbers divisible by 7 is:", div7)
print("List of numbers divisible by 8 is:", div8)
print("List of numbers divisible by 10 is:", div10)
    


List of numbers from 1 to 151 is: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151]
List of numbers divisible by 3 is: [3, 6, 9, 12, 15, 18, 21, 24, 27, 30, 33, 36, 39, 42, 45, 48, 51, 54, 57, 60, 63, 66, 69, 72, 75, 78, 81, 84, 87, 90, 93, 96, 99, 102, 105, 108, 111, 114, 117, 120, 123, 126, 129, 132, 135, 138, 141, 144, 147, 150]
List of numbers divisible by 4 is: [4, 8, 12, 16, 20, 24, 28, 32, 36

Given a list containing elements of different types (integers, strings, and floats), create three separate lists to store integers, strings, and floats respectively.

In [30]:
a = [1,'abc',3.12,True, 5, 'xyz', 7.89, False]
integer_list = []
float_list = []
string_list = []
bool_list = []
for i in range(len(a)):
    if type(a[i]) == int:
        integer_list.append(a[i])
    elif type(a[i]) == float:
        float_list.append(a[i])
    elif type(a[i]) == str:
        string_list.append(a[i])
    elif type(a[i]) == bool:
        bool_list.append(a[i]) 
print("List of integers is:", integer_list)
print("List of floats is:", float_list)
print("List of strings is:", string_list)
print("List of booleans is:", bool_list)

List of integers is: [1, 5]
List of floats is: [3.12, 7.89]
List of strings is: ['abc', 'xyz']
List of booleans is: [True, False]


What’s The Difference Between The Python append () and extend() Methods?

append () adds entire object as 1 element.
For Example 
a = [1,2,3] then if we append 3
a.append(3) it adds the element to the list so final output will be 
a = [1,2,3,3]
Or if we append ([3,4]) to the ad then it will become 
a = [1,2,3,[3,4]]
extend() takes the content and adds then individually

for Example, 
Or if we append([3,4]) to the ad then it will become 
a = [1,2,3,3,4]

it iterates and then add each element to the list similar to string if we extend ‘Hi’ to the list then it will add as a = [1,2,3, h , i ] because string is iterable.


Write a Python program to append a list to the second list

In [32]:
a = [1,2,3,4,5,6,7,8,9]
b = ['hi',1,2.3,True,'hello',False,3.14]

a.extend(b)

print(a)

[1, 2, 3, 4, 5, 6, 7, 8, 9, 'hi', 1, 2.3, True, 'hello', False, 3.14]


Write a Python program to find the third-largest number in a list

In [37]:
a = [1,23,56,9,32,]

for i in range(len(a)):
    a.sort()
    a.reverse()

print(f"3rd Largest element in array is : {a[2]}")

3rd Largest element in array is : 23


Write a Python program to check whether a list contains a sublist

In [40]:
a = [1,2,3,4,5,6,7,8,9]
b = [3,5]
found = False
for i in range(len(a) - (len(b) + 1)):
    if a[i:i+len(b)] == b:
        found = True
        break
print("Sublist found" if found else "sublist not found")


sublist not found


Write a Python program to generate all sublists of a list

In [43]:
a = [1,2,3]
b = []
for i in range(len(a)):
    for j in range(i+1,len(a)+1):
       print(a[i:j])

[1]
[1, 2]
[1, 2, 3]
[2]
[2, 3]
[3]


Write a Python program to find common items from two lists

In [46]:
a= [10,20,30,40,50,60,70,80,90,100]
b = [40,20,30]

common = []
for i in a:
    if i in b:
        common.append(i)
print("Common elements in both lists are:", common)

Common elements in both lists are: [20, 30, 40]


How to flatten a list in Python?

We can use extend function to flatter an array but every element should be iterable and if a single element is integer then the extend will fail so another method is to use a nested loop to find if the individual items and then copy to an array and print.

How to sort a list in ascending and descending order without using the sort function?

In [48]:
a = [32,3412,213,123,43,12,324,546]

for i in range(len(a)):
    smallest = i
    for j in range(i+1, len(a)):
        if a[j] < a[smallest]:
            smallest = j
    a[i],a[smallest] = a[smallest],a[i]
print("Sorted list is:", a)


Sorted list is: [12, 32, 43, 123, 213, 324, 546, 3412]


How to sort a tuple?

There are 2 ways to sort a tuple either manually create a list and then by finding the lowest elements and then copy them to a new list and then convert to a tuple. 2nd thing is that I can use a sorted method to sort the tuple and get a new list and then explicitly convert the list to a tuple.

In [49]:
a = (32,345,45,123,45)
b =tuple(sorted(a))

print(a)

(32, 345, 45, 123, 45)


Write a Python program to convert a list of multiple integers into a single integer ● [11, 33, 50] >>> 113350

In [53]:
a = [1,2,3,4,5,6,7,8,9]
b =""
for i in range(len(a)):
    b += str(a[i]) 
print(int(b))
print(type(int(b)))

123456789
<class 'int'>


What is the difference between del and clear?

Clear() functions clears the entire list but the list still exists whereas the del function can be used to delete a specific element or a range or full list once deleted the list wont be existing.

What is the difference between remove and pop?

Remove method is used to specifically remove an object from a list and it can cause a value error if the value is not in the list. Whereas pop is used to remove an element based on the position or index and it returns the removed element and if we use without parameters then it remove the last element from the list.

What is the difference between indexing and Slicing?

Indexing basically gets a single element whereas slicing we can specify a range which returns the elements in the range specified

What is the difference between sort and sorted?

Sort basically sorts the original list and returns none whereas sorted returns a new sorted list.

What is the difference between reverse and reversed function?

Same as sorted reverse will modify the original list in revered manner and reversed is an inbuilt function that returns a new list with revered elements. 

What is the difference between copy and deep copy?

If we use normal a = b then it shares a common list. but when we do copy() like a.copy ()  it copies the list but when values are modified inside a copy then if it has nested lists it will change the value inside the nested lists.  But whereas when we use deep copy it copies with the nested list inside . so if we update a value in the copy in the nested lists the values in the original list are not affected.

How to check whether the list is empty or not?

There as different methods to check this basic thing we can use len() function to the length and see if it is empty or we can use if not or if to check this as it returns true or false.

How to concatenate two lists?

 We can use + operator to join the list and create a new list and also extend method to concatenate 2 list into a single one which merges 2 list

Given a list of integers, create a new list where each element is replaced by the next largest number in the original list. If there is no larger number, replace the element with -1. ● input_list = [4, 2, 6, 8, 1, 2] ● output_list = [6, 6, 8, -1, 2, -1]

In [60]:
input_list = [4, 2, 6, 8, 1, 2] 
for i in range(len(input_list)):
    found = False
    for j in range(i+1,len(input_list)):
        if (input_list[i] < input_list[j]):
            found = True
            input_list[i] = input_list[j]
            break

    if not found:
        found = False
        input_list[i] = -1
print(input_list)


[6, 6, 8, -1, 2, -1]


Write a function to reverse the elements within each tuple in a list of tuples. ● tuples = [(1, 2), (3, 4), (5, 6)] ● Output: [(2, 1), (4, 3), (6, 5)]

In [61]:
input_list = [(1, 2), (3, 4), (5, 6)] 
new_list = []

for i in  range(len(input_list)):
    a = tuple(reversed(input_list[i]))
    new_list.append(a)

print("Original list of tuples:", input_list)
print("List of tuples after reversing:", new_list)


Original list of tuples: [(1, 2), (3, 4), (5, 6)]
List of tuples after reversing: [(2, 1), (4, 3), (6, 5)]


Write a function to add corresponding elements of tuples in two lists of tuples ● list1 = [(1, 2), (3, 4)] ● list2 = [(5, 6), (7, 8)] ● Output: [(6, 8), (10, 12)]

In [64]:
list1 = [(1, 2), (3, 4)] 
list2 = [(5, 6), (7, 8)] 

result = []
for i in  range(len(list1)):
    sum = (list1[i][0] + list2[i][0], list1[i][1] + list2[i][1])
    result.append(sum)
print(result)

[(6, 8), (10, 12)]
